processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -948,8 +949,21 @@ func (c *WatchClient) shouldIgnorePod(pod *api_v1.Pod) bool {
 
 func selectorsFromFilters(filters Filters) (labels.Selector, fields.Selector, error) {
 	labelSelector := labels.Everything()
+	singleValueSelections := []selection.Operator{
```

Comment:
lets make this a map so we can do constant time lookup instead of `slices.Contains`. It can be instantiated as a var outside the function.